lib/connection-manager.js

```diff
@@ -63,142 +81,62 @@ module.exports.connectLogServer = function(config, programName) {
         , reconnection: config.LOGDNA_RECONNECT
         , perMessageDeflate: !!config.COMPRESS
     };
+    var firstrun = true;
     var proxy = getProxyFromURI(url.parse(logurl));
     if (proxy) options.agent = new HttpsProxyAgent(url.parse(proxy));
-
-    var firstrun = true;
     debug(logurl);
-    var socket = new WebSocket(logurl, options);
-    linebuffer.setSocket(socket);
-    debug(socket.options);
-    return new Promise(resolve => {
-        debug('connecting to websocket');
-        socket.on('open', function() {
-            log('Connected to ' + config.LOGDNA_LOGHOST + ':' + config.LOGDNA_LOGPORT + ' (' + socket._socket.remoteAddress + ')' + (config.LOGDNA_LOGSSL ? ' (SSL)' : '') + (proxy ? ' via ' + proxy : ''));
-
-            if (firstrun) {
-                firstrun = false;
-
-                // start streaming logs from logdir(s) on startup
-                fileUtils.streamAllLogs(config);
-
-                if (os.platform() === 'win32' && config.winevent) {
-                    if (typeof config.winevent === 'string') {
-                        config.winevent = config.winevent.split(',');
+    var req = linebuffer.setConnection(logurl, options);
+    return new Promise((resolve) => {
+        setInterval(() => {
```

Comment:
usually you want to avoid using setInterval's, also is it possible that you have a compounding problem when that async process is constantly retrying?

ie. if the service is down and you area always retrying, it's possible that you kick off many simultaneous req.post's even though all of them are failing.